src/Solr/CwpSolr.php

```diff
@@ -111,7 +111,14 @@ public static function options_for_local($options)
                     : BASE_PATH . '/.solr',
                 'remotepath' => Environment::getEnv('SOLR_REMOTE_PATH')
                     ? Environment::getEnv('SOLR_REMOTE_PATH')
-                    : null
+                    : null,
+                'port' => Environment::getEnv('SOLR_INDEXSTORE_PORT')
+                    ? Environment::getEnv('SOLR_INDEXSTORE_PORT')
+                    : (
+                        Environment::getEnv('SOLR_PORT')
+                            ? Environment::getEnv('SOLR_PORT')
+                            : 8983
+                    )
```

Comment:
I don't know whether the CWP team wants this configurability to be added, but from a code perspective might I suggest you write a method like `getIndexStorePort()` or something to tidy this up a little?

Reply:
It would be the only exception having a separate function, but why not.

This is really useful for local CWP recipe-based dev using the webdav mode, unless CWP exposes their custom POST handler. The file mode is not always available for local dev where having a shared folder between the web sever and solr server may not be available.

Reply:
@robbieaverill note that it's within the `options_for_local` method, which doesn't apply within the CWP platform, so it's only helping local dev.

Reply:
We can use PHP 7 syntax in CWP (new releases) now right? Maybe you could do `Environment::getEnv('SOLR_INDEX_PORT') ?? Environment::getEnv('SOLR_PORT') ?? 8983` - that's a bit easier to read at a glance

Reply:
Yes, this module requires `cwp/cwp` which requires `cwp/cwp-core` which requires PHP > 7.1.

Updated the syntax and force-pushed.